### PR TITLE
fix(next): omit "fetch" from options

### PR DESCRIPTION
**Pull request by withyellow**
## 🎯 Changes

Use `fetch` from options when it's defined.
In the current implementation, `opts.fetch` it typed but never gets used.

## ✅ Checklist

- [x] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [x] If necessary, I have added documentation related to the changes made.
- [ ] I have added or updated the tests related to the changes made.

### CODE REVIEW

File: packages/next/src/app-dir/links/nextHttp.ts [hunk 1]
```diff
@@ -33,11 +33,12 @@ export function experimental_nextHttpLink<
       const revalidate = requestRevalidate ?? opts.revalidate ?? false;
 
       const linkFactory = opts.batch ? httpBatchLink : httpLink;
+      const fetchFn = opts.fetch ?? fetch;
       const link = linkFactory({
         headers: opts.headers as any,
         url: opts.url,
         fetch: (url, fetchOpts) => {
-          return fetch(url, {
+          return fetchFn(url, {
```
Comment (kamilogorek) [inline packages/next/src/app-dir/links/nextHttp.ts:41]: There's `getFetch` util from `@trpc/client` that you can use here.

Comment (juliusmarminge) [reply packages/next/src/app-dir/links/nextHttp.ts:41]: We need next’s polyfilled fetch in order to use their cache system though right? So I dont see why you would provide a custom fetch function when using this link

Comment (withyellow) [reply packages/next/src/app-dir/links/nextHttp.ts:41]:
``` typescript
next: {
  revalidate,
  tags: [cacheTag],
}
```

Maybe take more control of the `next` params.

Or just omit  `fetch` from `opts`.

**Comment by withyellow** [comment]
The main problem is that there's a `opts.fetch` type definition now, but it's not really being used in requests, which can be misleading. I think we should either use `opts.fetch` or omit it from `opts`.

File: packages/next/src/app-dir/links/nextHttp.ts (hunk 1 above)
Comment (juliusmarminge) [reply packages/next/src/app-dir/links/nextHttp.ts]: I’d say omit it from types, unless you can provide a concrete example of when you’d want to overwrite the next options (as modifying the cachetag will have to be changed in the revalidate() too which is bound to the client and not the link)

Comment (withyellow) [reply packages/next/src/app-dir/links/nextHttp.ts]: Done.

**Review by juliusmarminge — APPROVED** [review]

**Pull request merged**